File: mobile/ios/KompaniyaAnticor/Utilities/Theme.swift
import SwiftUI

/// Veb ilovaning rang palitrasi bilan mos (login.html/admin.html'dagi
/// --azure/--teal/--coral CSS o'zgaruvchilari).
enum AppColors {
    static let azure = Color(red: 0x00 / 255, green: 0x44 / 255, blue: 0x91 / 255)
    static let teal = Color(red: 0x1D / 255, green: 0x6F / 255, blue: 0xBF / 255)
    static let coral = Color(red: 0xFF / 255, green: 0x6B / 255, blue: 0x6B / 255)
    static let bg = Color(red: 0xF1 / 255, green: 0xF2 / 255, blue: 0xF4 / 255)
    static let bgDeep = Color(red: 0xE4 / 255, green: 0xE6 / 255, blue: 0xE9 / 255)
    static let text = Color(red: 0x0B / 255, green: 0x25 / 255, blue: 0x45 / 255)
    static let textDim = Color(red: 0x4E / 255, green: 0x6A / 255, blue: 0x88 / 255)
    static let cardBorder = Color(red: 0x0B / 255, green: 0x25 / 255, blue: 0x45 / 255).opacity(0.14)
}

File: mobile/flutter/lib/theme.dart
import 'package:flutter/material.dart';

/// Veb ilovaning rang palitrasi bilan mos (login.html/admin.html'dagi
/// --azure/--teal/--coral CSS o'zgaruvchilari).
class AppColors {
  static const azure = Color(0xFF004491);
  static const teal = Color(0xFF1D6FBF);
  static const coral = Color(0xFFFF6B6B);
  static const bg = Color(0xFFF1F2F4);
  static const bgDeep = Color(0xFFE4E6E9);
  static const text = Color(0xFF0B2545);
  static const textDim = Color(0xFF4E6A88);
  static const cardBorder = Color(0x230B2545);
}

ThemeData buildAppTheme() {
  final base = ThemeData(
    useMaterial3: true,
    colorScheme: ColorScheme.fromSeed(
      seedColor: AppColors.azure,
      primary: AppColors.azure,
      secondary: AppColors.teal,
      error: AppColors.coral,
      brightness: Brightness.light,
    ),
    scaffoldBackgroundColor: AppColors.bg,
  );

  return base.copyWith(
    appBarTheme: const AppBarTheme(
      backgroundColor: AppColors.bg,
      foregroundColor: AppColors.text,
      elevation: 0,
      centerTitle: false,
      titleTextStyle: TextStyle(color: AppColors.text, fontSize: 18, fontWeight: FontWeight.w700),
    ),
    cardTheme: CardThemeData(
      color: Colors.white,
      elevation: 0,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16),
        side: const BorderSide(color: AppColors.cardBorder),
      ),
    ),
    elevatedButtonTheme: ElevatedButtonThemeData(
      style: ElevatedButton.styleFrom(
        backgroundColor: AppColors.azure,
        foregroundColor: Colors.white,
        padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 14),
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
        textStyle: const TextStyle(fontWeight: FontWeight.w700, fontSize: 14),
      ),
    ),
    inputDecorationTheme: InputDecorationTheme(
      filled: true,
      fillColor: Colors.white,
      border: OutlineInputBorder(
        borderRadius: BorderRadius.circular(12),
        borderSide: const BorderSide(color: AppColors.cardBorder),
      ),
      enabledBorder: OutlineInputBorder(
        borderRadius: BorderRadius.circular(12),
        borderSide: const BorderSide(color: AppColors.cardBorder),
      ),
      focusedBorder: OutlineInputBorder(
        borderRadius: BorderRadius.circular(12),
        borderSide: const BorderSide(color: AppColors.azure, width: 1.6),
      ),
      contentPadding: const EdgeInsets.symmetric(horizontal: 14, vertical: 14),
    ),
  );
}
